dali/kernels/imgproc/convolution/cutlass/device/gemm.h

```diff
@@ -155,72 +159,97 @@ namespace device {
-                                          ElementAccumulator_>::kStages,
+    int Stages = DefaultConvConfiguration<OperatorClass_, ArchTag_,
+                                          select_A_t<InnerConv, ElementIn_, ElementWindow_>,
+                                          select_B_t<InnerConv, ElementIn_, ElementWindow_>,
+                                          ElementOut_, ElementAccumulator_>::kStages,
     /// Access granularity of A matrix in units of elements
-    int AlignmentA = DefaultGemmConfiguration<OperatorClass_, ArchTag_, ElementA_, ElementB_,
-                                              ElementC_, ElementAccumulator_>::kAlignmentA,
+    int AlignmentA = DefaultConvConfiguration<OperatorClass_, ArchTag_,
+                                              select_A_t<InnerConv, ElementIn_, ElementWindow_>,
+                                              select_B_t<InnerConv, ElementIn_, ElementWindow_>,
+                                              ElementOut_, ElementAccumulator_>::kAlignmentA,
     /// Access granularity of B matrix in units of elements
-    int AlignmentB = DefaultGemmConfiguration<OperatorClass_, ArchTag_, ElementA_, ElementB_,
-                                              ElementC_, ElementAccumulator_>::kAlignmentB,
+    int AlignmentB = DefaultConvConfiguration<OperatorClass_, ArchTag_,
+                                              select_A_t<InnerConv, ElementIn_, ElementWindow_>,
+                                              select_B_t<InnerConv, ElementIn_, ElementWindow_>,
+                                              ElementOut_, ElementAccumulator_>::kAlignmentB,
     /// If true, kernel supports split-K with serial reduction
     bool SplitKSerial = false,
     /// Operation performed by GEMM
-    typename Operator_ = typename DefaultGemmConfiguration<
-        OperatorClass_, ArchTag_, ElementA_, ElementB_, ElementC_, ElementAccumulator_>::Operator,
+    typename Operator_ = typename DefaultConvConfiguration<
+        OperatorClass_, ArchTag_,
+        select_A_t<InnerConv, ElementIn_, ElementWindow_>,
+        select_B_t<InnerConv, ElementIn_, ElementWindow_>,
+        ElementOut_, ElementAccumulator_>::Operator,
     /// Whether Beta is zero or not
     bool IsBetaZero = false>
-class Gemm {
+class Conv {
  public:
-  using ElementA = ElementA_;
-  using LayoutA = LayoutA_;
-  using TensorRefA = TensorRef<ElementA const, LayoutA>;
-  using ElementB = ElementB_;
-  using LayoutB = LayoutB_;
-  using TensorRefB = TensorRef<ElementB const, LayoutB>;
-  using ElementC = ElementC_;
-  using LayoutC = LayoutC_;
-  using TensorRefC = TensorRef<ElementC const, LayoutC>;
-  using TensorRefD = TensorRef<ElementC, LayoutC>;
+  using ElementIn = ElementIn_;
+  using ElementCastIn = ElementCastIn_;
+  using LayoutIn = LayoutIn_;
+  using ElementWindow = ElementWindow_;
+  using ElementCastWindow = ElementCastWindow_;
```

Comment:
"ElementWindow" sounds more like a "window for elements", not "element of a window". If you're attached to these names, they can stay, but I'm just saying that the name had me looking at code to get the meaning.
WindowElement, WindowLayout, InElement, InLayout, etc. would be more natural.

Reply:
It probably would be, but this is the convention used throughout cutlass and I stuck with it.

Reply:
If you have some better Element[Something] suggestion, I'm open to it.